src/main/java/com/pinterest/secor/uploader/Uploader.java

```diff
@@ -308,16 +319,15 @@ protected void checkTopicPartition(TopicPartition topicPartition, boolean forceU
                            isRequiredToUploadAtTime(topicPartition);
         }
         if (shouldUpload) {
-            long newOffsetCount = mZookeeperConnector.getCommittedOffsetCount(topicPartition);
+            long newOffsetCount = getRemoteComittedOffsetCount(topicPartition);
```

Comment:
this could be anti-intuitive:
this change adds a branching to use zk or kafka depending on which one is selected. However, we use dual commit, so this is going to use zookeeper offset, which is the current behaviour. 

The problem is addressed by other code change and is a side effect. See the next comment